addons/portal/static/src/js/portal_share_document.js

```diff
@@ -0,0 +1,110 @@
+odoo.define('portal.share_document', function (require) {
+'use strict';
+
+var core = require('web.core');
+var data_manager = require('web.data_manager');
+var SystrayMenu = require('web.SystrayMenu');
+var WebClient = require('web.WebClient');
+var Widget = require('web.Widget');
+
+var _t = core._t;
+
+var ProtalShareDoc = Widget.extend({
+    template: 'portal.sharing_icon',
+    events: {
+        "click": "_onClick",
+    },
+    xmlDependencies: ['/portal/static/src/xml/portal_share_document.xml'],
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    /**
+     * Update the system tray icon for share document, based on view type hides/shows icon
+     * Show Share Icon only if 1) action context has share_icon True 2) form view of portal.mixin inherited object
+     */
+    update: function (tag, descriptor, widget) {
```

Comment:
almost all of the logic here should be in the form view/other views, and not here.  The web client should not be aware of the way sub components are implemented.

The form view could for example trigger an event 'share_action' with the relevant details on core.bus, and this widget could listen on it

Reply:
@ged-odoo 

our icon is on the system tray and we also need to hide the icon when we switch from form view to other views(we may trigger based on switch_view but while navigating through the breadcrum switch_view is not called) and also for discuss client action we need icon on system tray to invite new user. so the only way is to include in webclient and trigger from current_action_updated. let me know if we can have any better solution for this? Thanks